File: 2-dockerize_nestjs/libs/contract/src/dto/api-error.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class ApiErrorDto {
  @ApiProperty({ example: 'a1b2c3d4e5f6' }) id: string;
  @ApiProperty({ example: 500 }) httpStatusCode: number;
  @ApiProperty({ example: '' }) code: string;
  @ApiProperty({ example: '/api/point-of-interest' }) links: string;
  @ApiProperty({ example: 'An error occurred in the API.' }) title: string;
  @ApiProperty({ example: 'Something went wrong.' }) message: string;
  @ApiProperty({ example: '' }) detail: string;
}

File: 0-preview/common/mytravels.contract/Dtos/ApiErrorDto.cs
using System.Diagnostics.CodeAnalysis;

namespace mytravels.contract.Dtos;

[ExcludeFromCodeCoverage]
public class ApiErrorDto
{
    public string Id { get; set; }
    public int HttpStatusCode { get; set; }
    public string Code { get; set; }
    public string Links { get; set; }
    public string Title { get; set; }
    public string Message { get; set; }
    public string Detail { get; set; }
}
